### ISSUE TITLE
io.rancher.container.network - label does not result in container getting a managed ip.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#6570]
Rancher Version: Build from master

Steps to reproduce the problem:
Create a container from host using label - 'io.rancher.container.network': 'true' ,
 We expect container to get create in "managed" network with ipaddress in 10.42.x.x .
Instead the container is given - "172.17.x.x" address.

```
_________________________ test_native_managed_network __________________________

socat_containers = None, client = <cattle.Client object at 0x1100e2190>
native_name = 'native-test-143527', pull_images = None

    def test_native_managed_network(socat_containers, client, native_name,
                                    pull_images):
        docker_client = get_docker_client(host(client))
        docker_container = docker_client. \
            create_container(NATIVE_TEST_IMAGE,
                             name=native_name,
                             labels={'io.rancher.container.network': 'true'})
        container, docker_container = start_and_wait(client, docker_container,
                                                     docker_client,
                                                     native_name)
    
        assert container.externalId == docker_container['Id']
        assert container.state == 'running'
>       assert container.primaryIpAddress != docker_container['NetworkSettings'][
            'IPAddress']
E       assert '172.17.0.4' != '172.17.0.4'
E        +  where '172.17.0.4' = {'tty': False, 'registryCredentialId': None, 'labels': {'io.rancher.container.volume_cleanup_strategy': u'none', 'io.r...nd': [u'nodejs', u'/app/app.js'], 'serviceIds': None, 'deploymentUnitUuid': None, 'createIndex': None, 'dnsSearch': []}.primaryIpAddress

/Users/sangeethahariharan1/cattle/validation/validation-tests/tests/v2_validation/cattlevalidationtest/core/test_native_docker.py:149: AssertionError
=========== 11 tests deselected by '-k test_native_managed_network' ============
=================== 1 failed, 11 deselected in 47.11 seconds ===================

Process finished with exit code 0

socat_containers = None, client = <cattle.Client object at 0x1100e2190>
native_name = 'native-test-143527', pull_images = None

    def test_native_managed_network(socat_containers, client, native_name,
                                    pull_images):
        docker_client = get_docker_client(host(client))
        docker_container = docker_client. \
            create_container(NATIVE_TEST_IMAGE,
                             name=native_name,
                             labels={'io.rancher.container.network': 'true'})
        container, docker_container = start_and_wait(client, docker_container,
                                                     docker_client,
                                                     native_name)
    
        assert container.externalId == docker_container['Id']
        assert container.state == 'running'
>       assert container.primaryIpAddress != docker_container['NetworkSettings'][
            'IPAddress']
E       assert '172.17.0.4' != '172.17.0.4'
E        +  where '172.17.0.4' = {'tty': False, 'registryCredentialId': None, 'labels': {'io.rancher.container.volume_cleanup_strategy': u'none', 'io.r...nd': [u'nodejs', u'/app/app.js'], 'serviceIds': None, 'deploymentUnitUuid': None, 'createIndex': None, 'dnsSearch': []}.primaryIpAddress

/Users/sangeethahariharan1/cattle/validation/validation-tests/tests/v2_validation/cattlevalidationtest/core/test_native_docker.py:149: AssertionError
```

**Comment by sangeethah** [comment]
Following automated test also fails for the same reason:

```
____________________________ test_native_ip_inject _____________________________

client = <cattle.Client object at 0x102ff6e50>, socat_containers = None
native_name = 'native-test-226290', pull_images = None

    def test_native_ip_inject(client, socat_containers, native_name,
                              pull_images):
        docker_client = get_docker_client(host(client))
        docker_container = docker_client. \
            create_container(NATIVE_TEST_IMAGE,
                             name=native_name,
                             labels={'io.rancher.container.network': 'true'},
                             tty=True,
                             stdin_open=True,
                             detach=True,
                             command=['/bin/bash', '-c', 'until $(ip addr show | '
                                                         'grep -q 10.42); '
                                                         'do sleep 1 && echo .; '
                                                         'done; ip addr show'])
        rancher_container, _ = start_and_wait(client, docker_container,
                                              docker_client, native_name)
>       assert_ip_inject(client.reload(rancher_container))

/Users/sangeethahariharan1/cattle/validation/validation-tests/tests/v2_validation/cattlevalidationtest/core/test_native_docker.py:261: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

container = {'tty': True, 'registryCredentialId': None, 'labels': {'io.rancher.container.volume_cleanup_strategy': u'none', 'io.ra... && echo .; done; ip addr show'], 'serviceIds': None, 'deploymentUnitUuid': None, 'createIndex': None, 'dnsSearch': []}

    def assert_ip_inject(container):
        ip = container.primaryIpAddress
        logs = container.logs()
        conn = ws.create_connection(logs.url + '?token=' + logs.token, timeout=10)
        count = 0
        found_ip = False
        while count <= 100:
            count += 1
            try:
                result = conn.recv()
                if ip in result:
                    found_ip = True
                    break
            except ws.WebSocketConnectionClosedException:
                break
>       assert found_ip
E       assert False

/Users/sangeethahariharan1/cattle/validation/validation-tests/tests/v2_validation/cattlevalidationtest/core/test_container.py:153: AssertionError
============== 11 tests deselected by '-k test_native_ip_inject' ===============
================== 1 failed, 11 deselected in 151.59 seconds ===================

Process finished with exit code 0

client = <cattle.Client object at 0x102ff6e50>, socat_containers = None
native_name = 'native-test-226290', pull_images = None

    def test_native_ip_inject(client, socat_containers, native_name,
                              pull_images):
        docker_client = get_docker_client(host(client))
        docker_container = docker_client. \
            create_container(NATIVE_TEST_IMAGE,
                             name=native_name,
                             labels={'io.rancher.container.network': 'true'},
                             tty=True,
                             stdin_open=True,
                             detach=True,
                             command=['/bin/bash', '-c', 'until $(ip addr show | '
                                                         'grep -q 10.42); '
                                                         'do sleep 1 && echo .; '
                                                         'done; ip addr show'])
        rancher_container, _ = start_and_wait(client, docker_container,
                                              docker_client, native_name)
>       assert_ip_inject(client.reload(rancher_container))

/Users/sangeethahariharan1/cattle/validation/validation-tests/tests/v2_validation/cattlevalidationtest/core/test_native_docker.py:261: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

container = {'tty': True, 'registryCredentialId': None, 'labels': {'io.rancher.container.volume_cleanup_strategy': u'none', 'io.ra... && echo .; done; ip addr show'], 'serviceIds': None, 'deploymentUnitUuid': None, 'createIndex': None, 'dnsSearch': []}

    def assert_ip_inject(container):
        ip = container.primaryIpAddress
        logs = container.logs()
        conn = ws.create_connection(logs.url + '?token=' + logs.token, timeout=10)
        count = 0
        found_ip = False
        while count <= 100:
            count += 1
            try:
                result = conn.recv()
                if ip in result:
                    found_ip = True
                    break
            except ws.WebSocketConnectionClosedException:
                break
>       assert found_ip
E       assert False

/Users/sangeethahariharan1/cattle/validation/validation-tests/tests/v2_validation/cattlevalidationtest/core/test_container.py:153: AssertionError
```

**Comment by deniseschannon** [comment]
Will be addressed in plugin-manager v0.2.8 and cattle PR

**Comment by sangeethah** [comment]
Tested with v1.2.0-pre4-rc14 and rancher/plugin-manager:v0.2.9

Both test_native_ip_inject and test_native_managed_network test cases succeed.
